File: pinpointB/src/main/java/com/nxquar/pinpoint/service/InstituteService.java
package com.nxquar.pinpoint.service;

import com.nxquar.pinpoint.DTO.InstituteRequest;
import com.nxquar.pinpoint.DTO.MessageResponse;
import com.nxquar.pinpoint.DTO.branch.InstituteResponse;

import java.util.UUID;

public interface InstituteService {
    InstituteResponse getInstituteById(UUID id, String jwt);
    MessageResponse updateInstitute(InstituteRequest request, String jwt);
    MessageResponse deleteInstitute(String jwt);
}

File: pinpoint/lib/data/datasource/institute/institute_service.dart
import 'package:pinpoint/model/institute/institute.dart';
import 'package:pinpoint/model/institute/institute_request.dart';
import 'package:pinpoint/model/message_response.dart';
import 'package:pinpoint/resources/constant/string/appString.dart';
import 'package:retrofit/retrofit.dart';
import 'package:dio/dio.dart';

part 'institute_service.g.dart';

@RestApi(baseUrl: AppString.baseUrl)
abstract class InstituteApi {
  factory InstituteApi(Dio dio, {String baseUrl}) = _InstituteApi;

  @GET("/api/institute")
  Future<InstituteResponse> getInstitute();

  @PUT("/api/institute")
  Future<MessageResponse> updateInstitute(
    @Body() InstituteRequest request,
  );

  @DELETE("/api/institute")
  Future<MessageResponse> deleteInstitute(
  );
}
